include/host_queue.h

```diff
@@ -159,7 +160,12 @@ namespace detail {
 			MPI_Comm comm;
 			ctpl::thread_pool thread;
 
-			comm_thread(MPI_Comm comm, size_t n_threads) : comm(comm), thread(n_threads) {}
+			comm_thread(MPI_Comm comm, size_t n_threads) : comm(comm), thread(n_threads) {
```

Comment:
I'd propose to pass an "id" into here (which is simply `threads.size()`) which can then be prefixed to the worker name, i.e., `fmt::format("worker-{}.{}", id, i)`.

Reply:
Done.